modules/tinymce/src/core/main/ts/dom/ScrollIntoView.ts

```diff
@@ -117,47 +117,79 @@ const preserveWith = (editor: Editor, f: (startElement: SugarElement<Node>, endE
   editor.selection.setRng(rng);
 };
 
-const scrollToMarker = (editor: Editor, marker: MarkerInfo, viewHeight: number, alignToTop: boolean, doc?: SugarElement<Document>) => {
+const scrollToMarker = (editor: Editor, marker: MarkerInfo, viewHeight: number, alignToTop: boolean, doc?: SugarElement<Document>, topMargin?: boolean) => {
   const pos = marker.pos;
+  const viewWidth = Traverse.defaultView(doc ?? SugarElement.fromDom(editor.getDoc())).dom.innerWidth;
+  const viewportLeft = -editor.getBody().getBoundingClientRect().left;
   if (alignToTop) {
-    Scroll.to(pos.left, pos.top, doc);
+    if (topMargin) {
+      const targetOffset = viewHeight * 0.25;
+      const y = pos.top - targetOffset;
+      // If element is off screen to the left, scroll to slightly right of its position
+      if (pos.left < viewportLeft) {
+        Scroll.to(pos.left - 20, y, doc);
```

Comment:
So we are also altering the left position so the `topMargin` does special things to left position. I would expect that `topMargin` would just alter the top not other properties of the rect. And 20 pixels seems very arbitrary should this be a config in? This could be a config so `scrollIntoView(elm, { leftMargin: '20px' })` or maybe `scrollIntoView(elm, { leftMargin: 20 })`

Reply:
for this ticket at least we no longer modify the left or right scrolling.

We don't have any accurate data on how likely the editor will scroll horizontally, but this seems to be an issue we can put into the back log?